### DISCUSSION TITLE
multiple meters with sdcard logging

### DISCUSSION THREAD

**Discussion by gemu2015** [arendst/Tasmota#14218] [Show and tell]
just wanted to show what is possible with tasmota. this setup uses an ESP32 to collect 7 Meters in Total:

2 Modbus devices SDM530 and SDM630 for several solar inverters and wallbox
1 OBIS meter for solar input
1 SML meter for global house energy input and output
1 EBUS input for heater and solar water
1 Counter for water meter
1 serial OBIS from a special gas meter device which reads the meter through an ESP32 CAM
https://github.com/jomjol/AI-on-the-edge-device

these are 3 hardware and 2 software serial interfaces and one counter

All meters are displayed and logged on SD card every 15 Minutes and in an extra file at midnight.

these log files are emailed every month for backup

all meter instant values are distributed in the local net via global variables.

to give an idea how this works see the attached script.

on a central core2 i collect all the log files and global values and display extended infos like weekly or daily graphs. this makes the raspi obsolete that i used before.

![multimeter1](https://user-images.githubusercontent.com/11647075/147753842-a666ce04-8246-4490-9ec3-8a5ccb59b78d.jpg)

[color_script.pdf](https://github.com/arendst/Tasmota/files/7792979/color_script.pdf)

**Comment by barbudor** [comment]
Impressive @gemu2015 !

**Comment by arendst** [comment]
So all your Tasmota tweaks lead to this! Well done!
